cmd/ingester/app/consumer/consumer.go

```diff
@@ -67,25 +66,30 @@ func New(params Params) (*Consumer, error) {
 		internalConsumer:    params.InternalConsumer,
 		processorFactory:    params.ProcessorFactory,
 		deadlockDetector:    deadlockDetector,
-		partitionIDToState:  make(map[int32]*consumerState),
```

Comment:
What makes this unnecessary?

Reply:
partitionIDToState is only used for mock unit tests that fit sarama-cluster repositories

----
https://github.com/jaegertracing/jaeger/blob/main/cmd/ingester/app/consumer/consumer_test.go#L142

`func TestSaramaConsumerWrapper_start_Messages(t *testing.T) {`

<img width="868" alt="image" src="https://user-images.githubusercontent.com/8416556/226110531-f15067b5-be65-4cf3-a62a-8942785ec7d2.png">

---